### PR TITLE
2022.1.0 release

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by oleksandr-pavlyk** [comment]
I have verified locally that `conda build recipe -c conda-forge` works fine and the added test step of `dpcpp simple.cpp -o simple.exe` worked on Ubuntu.

Any insight into why it may be failing in the CI?

**Comment by napetrov** [comment]
dpcpp: error: linker command failed with exit code 1 (use -v to see invocation)

Something to do with toolchain on system. Might be it's due to OS/glibc/gcc versions in image in conda-forge CI.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -297,12 +297,20 @@ outputs:
...
+        - {{ compiler('cxx') }}
+      files:
+        - simple.cpp
       commands:
         - ls -A1 ${PREFIX}/bin/*         # [unix]
         - ls -A1 ${PREFIX}/include/*     # [unix]
         - ls -A1 ${PREFIX}/lib/*         # [unix]
         - dir %PREFIX%\Library\bin\*  # [win]
         - dir %PREFIX%\Library\lib\*  # [win]
+        - dpcpp --gcc-toolchain=$PREFIX --sysroot=$PREFIX/$BUILD/sysroot -target $BUILD simple.cpp -lpthread -o simple.exe  # [unix]
```
Comment (beckermr) [inline recipe/meta.yaml:311]: @isuruf Can you look at this? I think this line needs CXXFLAGS and LDFLAGS maybe?

```suggestion
        - dpcpp --gcc-toolchain=$PREFIX --sysroot=$PREFIX/$BUILD/sysroot -target $BUILD ${LDFLAGS} ${CXXFLAGS} simple.cpp -lpthread -o simple.exe  # [unix]
```

**Comment by beckermr** [comment]
Oh my bad. You need the cos7 sysroot. Sit tight.

**Comment by beckermr** [comment]
@isuruf @oleksandr-pavlyk I think we need to properly pin this compiler to require the cos7 sysroot. I don't know where precisely the best place is to do this.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

**Comment by oleksandr-pavlyk** [comment]
Thank you for your help, @beckermr !

**Comment by beckermr** [comment]
Happy to help! We still need to change the recipe to pin cos7 properly. Let's wait to merge until we work that out.

**Comment by beckermr** [comment]
OK @isuruf @oleksandr-pavlyk commit https://github.com/conda-forge/intel-compiler-repack-feedstock/pull/4/commits/2a28956d9867f80729b11495424087d0509c5a3b does the following:

1. I added an `__glibc >=2.7<3.0.a0` strong run export to the dpcpp compiler `impl` and wrapper package. 
2. I added a run-time constraint on the appropriate linux sysroot for the dpcpp compiler `impl` package.

This will ensure that the compiler is running with the cos7 sysroot always and that anything built with it gets the proper runtime virtual package constraint that will force the users of builds to have a cos7 or later system.

cc @conda-forge/core for viz and advice

**Comment by beckermr** [comment]
@oleksandr-pavlyk Is there documentation somewhere on what OS versions the dpcpp compiler supports for both itself and the executables it makes?

**Comment by oleksandr-pavlyk** [comment]
I suppose it is here: https://www.intel.com/content/www/us/en/developer/articles/system-requirements/intel-oneapi-dpcpp-system-requirements.html

> Supported minimum versions of distributed libraries that the compilers are built on: GCC - 7.3.0, BINUTILS- 2.30, GLIBC-2.27

**Comment by beckermr** [comment]
Awesome. Then the pinnings here make sense (and are probably too liberal). 

Let's wait a week or so and if nobody objects we will merge.

**Review by napetrov — APPROVED** [review]

File: recipe/meta.yaml [hunk 2]
```diff
@@ -297,12 +300,20 @@ outputs:
...
+        - {{ compiler('cxx') }}
+      files:
+        - simple.cpp
       commands:
         - ls -A1 ${PREFIX}/bin/*         # [unix]
         - ls -A1 ${PREFIX}/include/*     # [unix]
         - ls -A1 ${PREFIX}/lib/*         # [unix]
         - dir %PREFIX%\Library\bin\*  # [win]
         - dir %PREFIX%\Library\lib\*  # [win]
+        - dpcpp --gcc-toolchain=$PREFIX --sysroot=$PREFIX/$BUILD/sysroot -target $BUILD ${LDFLAGS} ${CXXFLAGS} simple.cpp -lpthread -o simple.exe  # [unix]
```
Comment (isuruf) [inline recipe/meta.yaml:314]:
```suggestion
        - dpcpp --gcc-toolchain=$PREFIX --sysroot=$PREFIX/$HOST/sysroot -target $HOST ${LDFLAGS} ${CXXFLAGS} simple.cpp -lpthread -o simple.exe  # [unix]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -297,12 +300,20 @@ outputs:
...
+        # Need for -I option is a bug in compiler activation script. Can be removed when fixed
+        - dpcpp simple.cpp -I%PREFIX%\include -o simple.exe  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml:316]: SYCL include files should go into `%PREFIX%\Library\include` instead of `%PREFIX%\include`

Comment (oleksandr-pavlyk) [reply recipe/meta.yaml:316]: Yes, understood. This fix will be applied in the compiler package in a future release at which time I will remove this kludge.

**Comment by oleksandr-pavlyk** [comment]
@napetrov I think this can be merged now. Thank you everyone for reviewing and helping get this into proper shape. It took a village.

**Pull request merged**